src/Client.php

```diff
@@ -369,7 +369,13 @@ private function applyOptions(RequestInterface $request, array &$options): Reque
         }
 
         if (isset($options['json'])) {
-            $options['body'] = Utils::jsonEncode($options['json']);
+            $json_type = 0;
+            if (isset($options['json_type'])) {
```

Comment:
At the very least it should probably be called `json_encoding_options`

Since the encoder here is not configurable, I could see the value having _another_ configuration for this.

However, if it's so important, users might set `body` themselves (though it comes with missing convenience of automatically setting the content-type).

🤷‍♀️

Reply:
users set body themselves with json_encode with json_type then could't use optioins['json'].